### PR TITLE
Yaml parser regression with comments and non-strings

### PR DESCRIPTION
**Pull request by alexpott**
| Q             | A
| ------------- | ---
| Branch?       | 3.3 
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | no
| Fixed tickets | #25786 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -429,7 +429,24 @@ private function doParse($value, $flags)
...
+                                $parsedLine = $line;
+                            } else {
+                                $parsedLine = Inline::parse($line, $flags, $this->refs);
+                            }
+
+                            // There is a special case where an empty array
+                            // followed by a comment causes us to try to parse
+                            // the value as a multi-line string. In this
+                            // instance, return the empty array
+                            if ([] === $parsedLine) {
```
Comment (alexpott) [inline src/Symfony/Component/Yaml/Parser.php]: This does not work. If I change the {} to 1.1 ie. float then we get the parse exception.

**Comment by alexpott** [comment]
I've just added a new test `\Symfony\Component\Yaml\Tests\ParserTest::testMultiLineStringLastResortParsing()` which shows that the `Inline::parse()` in the loop is a bad idea anyway.

The following yaml fails in 3.3 / 3.4 at the moment:
```
test:
  You can have things that don't look like strings here
  true
  yes you can
```
At the moment this produces an exception because the `Inline::parse()` parses the `true` to a boolean.

**Comment by xabbuh** [comment]
moving to the 3.4 milestone as the last bugfix release for 3.3 was published today

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
Thank you @alexpott.

**Pull request closed without merging**